x-pack/plugins/monitoring/server/lib/alerts/fetch_ccr_read_exceptions.ts

```diff
@@ -35,6 +35,11 @@ export async function fetchCCRReadExceptions(
                 },
               },
             },
+            {
+              term: {
+                'data_stream.dataset': 'elasticsearch.ccr',
+              },
+            },
```

Comment:
Would this still work with the legacy indices? If we have to check for either the `type` or `dataset`, how about we factor out a function like

```typescript
const createDatasetFilter = (legacyType: string, dataset: string) => ({
  bool: {
    should: [
      {
        term: {
          type: legacyType,
        },
      },
      {
        term: {
          'data_stream.dataset': dataset,
        },
      },
    ],
    minimum_should_match: 1,
  },
})
```

which could then be used in this and the other similar queries as in

```typescript
{
  filter: [
    // other clauses
    createDatasetFilter('logstash_stats', 'logstash.node_stats'),
    // other clauses
  ]
}
```

?

Reply:
Thanks @weltenwort .  I implemented this helper function.